Force re-updating the resolutions of DateField

anasmi opened the pull request:
When value in the DF is changed on the client and as the response to ValueChangeEvent value is changed back on server, the change is not propagated to the client.
Through the value is correctly updated on the server
Fix https://github.com/vaadin/framework/issues/11099

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework/11159)
<!-- Reviewable:end -->

Commit:
Force re-updating the resolutions

When value in the DF is changed on the client and as the response to ValueChangeEvent value is changed back on server, the change is not propagated to the client.
Through the value is correctly updated on the server
Fix https://github.com/vaadin/framework/issues/11099

Commit:
Re-introduce removed breaks

elmot reviewed (changes requested):
> Reviewed 1 of 3 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/vaadin/framework/11159)* status: 2 of 3 files reviewed, 1 unresolved discussion (waiting on @anasmi and @elmot)

___
*[uitest/src/main/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListener.java, line 14 at r1](https://reviewable.io/reviews/vaadin/framework/11159#-LLYkr271lpAvJGuKIxY:-LLYkr271lpAvJGuKIxZ:b-rmm7op) ([raw file](https://github.com/vaadin/framework/blob/699b4aac124918b50cc8904fb9d74f68acc909f9/uitest/src/main/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListener.java#L14)):*
> ```Java
>     @Override
>     protected void setup(VaadinRequest request) {
>         LocalDate sourceDate = LocalDate.now();
> ```

Why not some fixed dates?
There is one or two second during the day when the test is going to fail. Not a blocker.
___
*[uitest/src/main/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListener.java, line 15 at r1](https://reviewable.io/reviews/vaadin/framework/11159#-LLYkS0A2KeNCneXTJ3a:-LLYkS0A2KeNCneXTJ3b:b-auid4o) ([raw file](https://github.com/vaadin/framework/blob/699b4aac124918b50cc8904fb9d74f68acc909f9/uitest/src/main/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListener.java#L15)):*
> ```Java
>     protected void setup(VaadinRequest request) {
>         LocalDate sourceDate = LocalDate.now();
>         DateField df = new DateField("Date: ");
> ```

Better to have explicit locale set as well.


<!-- Sent from Reviewable.io -->

elmot reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/vaadin/framework/11159)* status: 2 of 3 files reviewed, 2 unresolved discussions (waiting on @anasmi and @elmot)

___
*[uitest/src/main/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListener.java, line 26 at r1](https://reviewable.io/reviews/vaadin/framework/11159#-LLYlVh8DaMCmI3nz6kf:-LLYlVh8DaMCmI3nz6kg:b-zcou5j) ([raw file](https://github.com/vaadin/framework/blob/699b4aac124918b50cc8904fb9d74f68acc909f9/uitest/src/main/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListener.java#L26)):*
> ```Java
> 
>         addComponent(df);
>         Button setV = new Button("Set date after the current", e -> {
> ```

I'd say it's better to use three different fixed dates - initially set, then explicitly set and one set by ValueChangeListener. And check two of them, one before the click and one after.


<!-- Sent from Reviewable.io -->

elmot reviewed (commented):
> Reviewed 1 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/vaadin/framework/11159)* status: all files reviewed, 2 unresolved discussions (waiting on @anasmi)



<!-- Sent from Reviewable.io -->

Commit:
Modify tests

Commit:
Modify tests

anasmi commented:
___
*[uitest/src/main/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListener.java, line 14 at r1](https://reviewable.io/reviews/vaadin/framework/11159#-LLYkr271lpAvJGuKIxY:-LLZQW0CCh_BYE7vuFPu:b-886gsx) ([raw file](https://github.com/vaadin/framework/blob/699b4aac124918b50cc8904fb9d74f68acc909f9/uitest/src/main/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListener.java#L14)):*
<details><summary><i>Previously, elmot (Ilia Motornyi) wrote…</i></summary><blockquote>

Why not some fixed dates?
There is one or two second during the day when the test is going to fail. Not a blocker.
</blockquote></details>

Fixed


<!-- Sent from Reviewable.io -->

anasmi commented:
___
*[uitest/src/main/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListener.java, line 26 at r1](https://reviewable.io/reviews/vaadin/framework/11159#-LLYlVh8DaMCmI3nz6kf:-LLZQdE0BjzJozO35D2l:b-8faz1q) ([raw file](https://github.com/vaadin/framework/blob/699b4aac124918b50cc8904fb9d74f68acc909f9/uitest/src/main/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListener.java#L26)):*
<details><summary><i>Previously, elmot (Ilia Motornyi) wrote…</i></summary><blockquote>

I'd say it's better to use three different fixed dates - initially set, then explicitly set and one set by ValueChangeListener. And check two of them, one before the click and one after.
</blockquote></details>

Changed to have three separate days 


<!-- Sent from Reviewable.io -->

Commit:
Add ZoneId to the test

anasmi commented:
___
*[uitest/src/main/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListener.java, line 15 at r1](https://reviewable.io/reviews/vaadin/framework/11159#-LLYkS0A2KeNCneXTJ3a:-LLZSKDB42zxwJMxdLmc:bi9quoc) ([raw file](https://github.com/vaadin/framework/blob/699b4aac124918b50cc8904fb9d74f68acc909f9/uitest/src/main/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListener.java#L15)):*
<details><summary><i>Previously, elmot (Ilia Motornyi) wrote…</i></summary><blockquote>

Better to have explicit locale set as well.
</blockquote></details>

Added ZoneId


<!-- Sent from Reviewable.io -->

elmot reviewed (changes requested):
> Reviewed 1 of 1 files at r4, 1 of 1 files at r5, 1 of 1 files at r7.
> *[Reviewable](https://reviewable.io/reviews/vaadin/framework/11159)* status: all files reviewed, 3 unresolved discussions (waiting on @elmot and @anasmi)

___
*[uitest/src/main/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListener.java, line 23 at r7](https://reviewable.io/reviews/vaadin/framework/11159#-LLZ_Io098Ia-B4_G81R:-LLZ_Io098Ia-B4_G81S:br9oat9) ([raw file](https://github.com/vaadin/framework/blob/7e271bd94ee5cd0ad6367ca892726778f55620e6/uitest/src/main/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListener.java#L23)):*
> ```Java
>         df.setId("dateField1");
>         df.setValue(initialValue);
>         df.setZoneId(ZoneId.of("Europe/Helsinki"));
> ```

Locale, I mean. Timezone should not affect the test but Locale can.
___
*[uitest/src/test/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListenerTest.java, line 18 at r5](https://reviewable.io/reviews/vaadin/framework/11159#-LLZ_nTEEUAlKlzqh1HW:-LLZ_nTEEUAlKlzqh1HX:bshym4k) ([raw file](https://github.com/vaadin/framework/blob/0852127b407cccc84a472071105acc6f807e93db/uitest/src/test/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListenerTest.java#L18)):*
> ```Java
>         openTestURL();
> 
>         DateTimeFormatter format = DateTimeFormatter.ofPattern("d.M.yyyy");
> ```

Also better to use explicit locale


<!-- Sent from Reviewable.io -->

elmot reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/vaadin/framework/11159)* status: all files reviewed, 3 unresolved discussions (waiting on @elmot and @anasmi)

___
*[uitest/src/main/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListener.java, line 15 at r1](https://reviewable.io/reviews/vaadin/framework/11159#-LLYkS0A2KeNCneXTJ3a:-LLZa2UH30dx5ZIPgw1y:b602d2a) ([raw file](https://github.com/vaadin/framework/blob/699b4aac124918b50cc8904fb9d74f68acc909f9/uitest/src/main/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListener.java#L15)):*
<details><summary><i>Previously, anasmi (Anastasia Smirnova) wrote…</i></summary><blockquote>

Added ZoneId
</blockquote></details>

Should not affect the test functionality


<!-- Sent from Reviewable.io -->

Commit:
Add Locale to the test

Commit:
Merge branch 'master' into #1109

anasmi commented:
___
*[uitest/src/main/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListener.java, line 15 at r1](https://reviewable.io/reviews/vaadin/framework/11159#-LLYkS0A2KeNCneXTJ3a:-LLd-bra1Bfo0GlHpvwG:bqfm9s2) ([raw file](https://github.com/vaadin/framework/blob/699b4aac124918b50cc8904fb9d74f68acc909f9/uitest/src/main/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListener.java#L15)):*
<details><summary><i>Previously, elmot (Ilia Motornyi) wrote…</i></summary><blockquote>

Should not affect the test functionality
</blockquote></details>

Removed and added locale


<!-- Sent from Reviewable.io -->

anasmi commented:
___
*[uitest/src/main/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListener.java, line 23 at r7](https://reviewable.io/reviews/vaadin/framework/11159#-LLZ_Io098Ia-B4_G81R:-LLd-eHzAxbrKX_k4GQo:bqscysf) ([raw file](https://github.com/vaadin/framework/blob/7e271bd94ee5cd0ad6367ca892726778f55620e6/uitest/src/main/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListener.java#L23)):*
<details><summary><i>Previously, elmot (Ilia Motornyi) wrote…</i></summary><blockquote>

Locale, I mean. Timezone should not affect the test but Locale can.
</blockquote></details>

Added locale


<!-- Sent from Reviewable.io -->

anasmi commented:
___
*[uitest/src/test/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListenerTest.java, line 18 at r5](https://reviewable.io/reviews/vaadin/framework/11159#-LLZ_nTEEUAlKlzqh1HW:-LLd-gi94oS0VhbT2RoI:bqscysf) ([raw file](https://github.com/vaadin/framework/blob/0852127b407cccc84a472071105acc6f807e93db/uitest/src/test/java/com/vaadin/tests/components/datefield/DateFieldResetValueWithinListenerTest.java#L18)):*
<details><summary><i>Previously, elmot (Ilia Motornyi) wrote…</i></summary><blockquote>

Also better to use explicit locale
</blockquote></details>

Added locale


<!-- Sent from Reviewable.io -->

elmot reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r8.
> *[Reviewable](https://reviewable.io/reviews/vaadin/framework/11159#-:-LLj2WQ48QDkMg5rvSSn:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->
